docs: update RA setup documentation (#6888)
rust-lang/rust-clippy: CONTRIBUTING.md

## Getting code-completion for rustc internals to work

As of [#6869][6869], [`rust-analyzer`][ra_homepage] can understand that Clippy uses compiler-internals
using `extern crate` when `package.metadata.rust-analyzer.rustc_private` is set to `true` in Clippys `Cargo.toml.`
You will required a `nightly` toolchain with the `rustc-dev` component installed.
Make sure that in the `rust-analyzer` configuration, you set
```
{ "rust-analyzer.rustcSource": "discover" }
```
and
```
{ "rust-analyzer.updates.channel": "nightly" }
```
You should be able to see information on things like `Expr` or `EarlyContext` now if you hover them, also
a lot more type hints.

> This will work with `rust-analyzer <VERSION>` shipped in nightly `<VERSION>` and later.

Author: This will need to be updated once the nightly that contains the next ra-sync is released in a couple of days/a week or so,

## Getting code-completion for rustc internals to work

### IntelliJ Rust
Unfortunately, [`IntelliJ Rust`][IntelliJ_rust_homepage] does not (yet?) understand how Clippy uses compiler-internals
using `extern crate` and it also needs to be able to read the source files of the rustc-compiler which are not
available via a `rustup` component at the time of writing.
To work around this, you need to have a copy of the [rustc-repo][rustc_repo] available which can be obtained via
`git clone https://github.com/rust-lang/rust/`.
Then you can run a `cargo dev` command to automatically make Clippy use the rustc-repo via path-dependencies
which `IntelliJ Rust` will be able to understand.

> Run `cargo dev ra_setup --repo-path <repo-path>` where `<repo-path>` is an absolute path to the rustc repo

Reviewer: ```suggestion
Run `cargo dev ra_setup --repo-path <repo-path>` where `<repo-path>` is a path to the rustc repo
```

I patched this a while ago, so that you can also use a relative path.

> Run `cargo dev ra_setup --repo-path <repo-path>` where `<repo-path>` is a path to the rustc repo

Reviewer: ```suggestion
Run `cargo dev ide_setup --repo-path <repo-path>` where `<repo-path>` is a path to the rustc repo
```
